src/mongoose_router_dynamic_domains.erl

```diff
@@ -27,10 +26,9 @@ filter(From, To, Acc, Packet) ->
 
 route(From, To, Acc, Packet) ->
     LDstDomain = To#jid.lserver,
-    case mongoose_domain_api:get_host_type(LDstDomain) of
-        {ok, _} ->
-            ejabberd_local:register_host(LDstDomain),
+    case mongoose_lazy_routing:maybe_add_domain_or_subdomain(LDstDomain) of
```

Comment:
We can't create bottlenecks like this. If someone sends a lot of messages to unknown domains, this will time out. We should verify the host type **before** any synchronous code is called.

Reply:
Makes sense, we can check ETS tables before running gen_server:call/2